python/ngen_conf/src/ngen/config/init_config/value_unit_pair.py

```diff
@@ -0,0 +1,79 @@
+import re
+from typing import TYPE_CHECKING, Any, Generic, List, Optional, Type, TypeVar, Union
+
+from pydantic import validator
+from pydantic.generics import GenericModel
+from typing_extensions import Self, override
+
+if TYPE_CHECKING:
+    from pydantic.typing import AbstractSetIntStr, DictStrAny, MappingIntStrAny
+
+T = TypeVar("T")
+V = TypeVar("V")
+U = TypeVar("U")
+
+
+_VAL_UNIT_RE = re.compile(r"(.*)\[(.*)\]")
```

Comment:
Not that I think we should now, but may want to consider more restrictive regex here in the future...especially considering captures which have spaces, non-standard characters, ect.  This may be impossible to generalize and too far out of scope though -- just a passing thought.

Reply:
Yeah I agree. I didn't and don't know at this point in time what appropriate constraints to enforce here, so I left it a little open. My thinking was we could always refine this if needed.